crates/cyclers/code_generation/accessor.rs

```diff
@@ -0,0 +1,387 @@
+                                token_stream.append(TokenTree::Ident(format_ident!("as_ref")))
+                            }
+                            ReferenceType::Mutable => {
+                                token_stream.append(TokenTree::Ident(format_ident!("as_mut")))
+                            }
+                        }
+                        token_stream.append(TokenTree::Group(Group::new(
+                            Delimiter::Parenthesis,
+                            TokenStream::default(),
+                        )));
+                    }
+                }
+
+                if !is_last_segment {
+                    token_stream.append(TokenTree::Punct(Punct::new('.', Spacing::Alone)));
+                    let next_segments_contain_optional = path
+                        .iter()
+                        .skip(index + 1)
+                        .any(|segment| segment.is_optional);
+                    let method_name = match next_segments_contain_optional {
+                        true => "and_then",
+                        false => "map",
+                    };
+                    token_stream.append(TokenTree::Ident(format_ident!("{}", method_name)));
+
+                    let mut new_token_stream_within_method = TokenStream::default();
+                    new_token_stream_within_method
+                        .append(TokenTree::Punct(Punct::new('|', Spacing::Alone)));
+                    new_token_stream_within_method
+                        .append(TokenTree::Ident(format_ident!("{}", segment.name)));
+                    new_token_stream_within_method
+                        .append(TokenTree::Punct(Punct::new('|', Spacing::Alone)));
+                    if !next_segments_contain_optional {
+                        new_token_stream_within_method
+                            .append(TokenTree::Punct(Punct::new('&', Spacing::Alone)));
+                        if let ReferenceType::Mutable = reference_type {
+                            new_token_stream_within_method
+                                .append(TokenTree::Ident(format_ident!("mut")));
+                        }
+                    }
+                    new_token_stream_within_method
+                        .append(TokenTree::Ident(format_ident!("{}", segment.name)));
+                    token_stream_within_method = Some(new_token_stream_within_method);
+                }
+            }
+        }
+
+        if let Some(token_stream_within_method) = token_stream_within_method.take() {
+            token_stream.append(TokenTree::Group(Group::new(
+                Delimiter::Parenthesis,
+                token_stream_within_method,
+            )));
+        }
+
+        token_stream
+    }
+
+    let path_contains_variable = path.iter().any(|segment| {
+        if segment.is_variable && segment.name != "cycler_instance" {
+            unimplemented!("only $cycler_instance is implemented");
+        }
```

Comment:
Since `any(...)` returns immediately when it finds the first `true`, this check is only executed for the first "variable" segment.

Reply:
Splitted the checking from the variable-containment check.